File: python/1.py
from typing import List


class Solution:
    # def twoSum(self, nums: List[int], target: int) -> List[int]:
    #     d = {}

    #     for index, value in enumerate(nums):
    #         if value in d:
    #             return [d[value], index]
    #         else:
    #             k = target - value
    #             d[k] = index
    #     return
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i, i_v in enumerate(nums):
            for j, j_v in enumerate(nums):
                # if i_v == j_v:
                #     continue
                if i_v + j_v == target:
                    # print([i, j])
                    return [i, j]


sol = Solution()

print(sol.twoSum(nums=[2, 7, 11, 15], target=9))

File: neetcode/1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> output(2);
        unordered_map<int, int> ht;

        for (int i = 0; i < nums.size(); ++i) {
            // check if the value in hashtable
            // before add into
            if (ht.find(nums[i]) == ht.end()) {
                ht[target-nums[i]] = i;
            }

            else {
                int index = ht[nums[i]];
                output[0] = index;
                output[1] = i;
            }
        }

        return output;
    }
};